crawl-ref/source/dat/defaults/menu_colours.txt

```diff
@@ -8,52 +8,52 @@ menu += darkgrey:(melded)
 menu += darkgrey:(inert)
 
 # Bad items
-menu += lightred:.*bad_item.*
+menu += lightred:.*\bbad_item\b.*
```

Comment:
I don't really understand the changes to this file, sorry. Can you explain?

Reply:
With PCRE, or with GNU's POSIX implementation, it means that they don't match options such as "bad_item_prompt". There's no equivalent to \b in the POSIX standard, so I've changed it to use space now.

It would be easier if these all used a tag, but this works for now.

Reply:
I think the fix here should be to prevent `menu_colour` from applying to this menu -- despite the name it is really designed for items, not menus in general, and the default values reflect that. Though it looks like this was done intentionally in `EGP_MenuEntry::highlight_colour` so maybe there's a reason I'm not seeing?

Reply:
> The docs say menu_colour covers a lot more than just inventory menus, so I thought it was reasonable to apply it here. I notice that the code doesn't match this, but I don't know why this is.

The docs are wrong :-). Or rather, they are written in a confusing way, from a time where there essentially were no menus that didn't have something to do with items (so I believe this wording would be most likely referring to e.g. stash tracking). I just don't see a great reason to apply regex highlighting to options menus, just like it doesn't apply to the game menu, or the macro menu.